sympy/calculus/finite_diff.py

```diff
@@ -21,24 +21,28 @@
 from sympy.core.compatibility import iterable, range
 
 
-def finite_diff_weights(order, x_list, x0):
+def finite_diff_weights(order, x_list, x0=0.):
```

Comment:
Is there a particular reason for using `x=0.` instead of `x=0`?
`0.` generates a floating point value, sometimes this messes up symbolic representations but perhaps 0.0 is fine.